File: UserBehaviorAnalysis/src/main/scala/com/shangbaishuyao/bean/UserBehavior.scala
package com.shangbaishuyao.bean

/**
 * 用户行为的样例类
 *
 * @param userId
 * @param itemId
 * @param categoryId
 * @param behavior
 * @param timestamp
 */
case class UserBehavior(
                         userId:Long,
                         itemId:Long,
                         categoryId:Int,
                         behavior:String,
                         timestamp:Long)

File: flink-1.12.0-Demo/src/main/java/com/shangbaishuyao/demo/bean/UserBehavior.java
package com.shangbaishuyao.demo.bean;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserBehavior {
    private Long userId;
    private Long itemId;
    private Integer categoryId;
    private String behavior;
    private Long timestamp;
}
